Tasks/InstallAppleCertificateV2/preinstallcert.ts

```diff
@@ -26,20 +26,29 @@ async function run() {
         let certPwd: string = tl.getInput('certPwd');
 
         // get the P12 details - SHA1 hash and common name (CN)
-        let p12Hash: string = await sign.getP12SHA1Hash(certPath, certPwd);
+        let { fingerprint, commonName, notBefore, notAfter } = await sign.getP12Properties(certPath, certPwd);
         // give user an option to override the CN as a workaround if we can't parse the certificate's subject.
-        let p12CN: string = tl.getInput('certSigningIdentity', false);
-        if (!p12CN) {
-            p12CN = await sign.getP12CommonName(certPath, certPwd);
+        let commonNameOverride: string = tl.getInput('certSigningIdentity', false);
+        if (commonNameOverride) {
+            commonName = commonNameOverride;
         }
 
-        if (!p12Hash || !p12CN) {
+        if (!fingerprint || !commonName) {
             throw tl.loc('INVALID_P12');
```

Comment:
This is not part of your change, but I'd like to see more of our throws converted to `throw new 'blah`.  It preserves the stacktrace.

Reply:
I added `new Error` since we're trending towards using it and I assume its a best practice. But the only difference I could spot as a pipelines user was that the error message gets preceded with 'Error:'. I didn't spot any stack traces. With sys.debug set to true:

```
2018-12-12T00:36:37.9713270Z ##[debug]P12 fingerprint: undefined
2018-12-12T00:36:37.9713450Z ##[debug]P12 common name (CN): iPhone Developer: Chris Sidi (7RZ3N927YF)
2018-12-12T00:36:37.9713590Z ##[debug]NotBefore: undefined
2018-12-12T00:36:37.9713780Z ##[debug]NotAfter: undefined
2018-12-12T00:36:37.9713910Z ##[debug]certSigningIdentity=null
2018-12-12T00:36:37.9727870Z ##[debug]task result: Failed
2018-12-12T00:36:37.9794770Z ##[error]Error: Unable to find the certificate SHA1 hash and common name (CN). Verify that the certificate is a valid P12.
2018-12-12T00:36:37.9805430Z ##[debug]Processed: ##vso[task.issue type=error;]Error: Unable to find the certificate SHA1 hash and common name (CN). Verify that the certificate is a valid P12.
2018-12-12T00:36:37.9827530Z ##[debug]Processed: ##vso[task.complete result=Failed;]Error: Unable to find the certificate SHA1 hash and common name (CN). Verify that the certificate is a valid P12.
2018-12-12T00:36:37.9828330Z ##[debug]secure file name for id 79e9a66a-d5cb-4458-8bda-e5a6070a7d38 = chrisid_iOSDev_Nov2018.p12
2018-12-12T00:36:37.9828460Z ##[debug]Agent.TempDirectory=/Users/vsts/agent/2.142.1/work/_temp
2018-12-12T00:36:37.9828700Z ##[debug]Absolute path for pathSegments: /Users/vsts/agent/2.142.1/work/_temp,chrisid_iOSDev_Nov2018.p12 = /Users/vsts/agent/2.142.1/work/_temp/chrisid_iOSDev_Nov2018.p12
2018-12-12T00:36:37.9828900Z ##[debug]Deleting secure file at: /Users/vsts/agent/2.142.1/work/_temp/chrisid_iOSDev_Nov2018.p12
2018-12-12T00:36:37.9829340Z ##[debug]rm -rf /Users/vsts/agent/2.142.1/work/_temp/chrisid_iOSDev_Nov2018.p12
2018-12-12T00:36:37.9829550Z ##[debug]removing file
```

![image](https://user-images.githubusercontent.com/7145717/49840344-fe5dc700-fd80-11e8-950a-c41feeba63ec.png)